src/java/org/apache/cassandra/cql3/functions/TimeFcts.java

```diff
@@ -65,6 +76,12 @@ public ByteBuffer execute(ProtocolVersion protocolVersion, List<ByteBuffer> para
             {
                 return type.now();
             }
+
+            @Override
+            public boolean isPure()
+            {
+                return false;
```

Comment:
nit: maybe add a comment explain why `now()` is not pure as it returns non-identical results for identical arguments.